docs/components/material.md

```diff
@@ -70,7 +70,7 @@ depending on the material type applied.
 | shader       | Which material to use. Defaults to the [standard material][standard]. Can be set to the [flat material][flat] or to a registered custom shader material. | standard      |
 | side         | Which sides of the mesh to render. Can be one of `front`, `back`, or `double`.                                                                    | front         |
 | transparent  | Whether material is transparent. Transparent entities are rendered after non-transparent entities.                                                | false         |
-| vertexColors | Whether to use vertex or face colors to shade the material. Can be one of `none`, `vertex`, or `face`.                                            | none          |
+| vertexColors | Whether to use vertex colors to shade the material. Can be one of `none` or `vertex`.                                            | none          |
```

Comment:
Should we probably rename to `vertexColorsEnabled` since it's now a boolean?

Reply:
Advantage of keeping `material="vertexColors:vertex;"` is that we don't break anyone's code. And this will be less issues created in this repo when people will update their projects to latest aframe. :)
I would prefer to keep `vertexColors` instead of `vertexColorsEnabled` having a name different than the THREE.Material property. Maybe we can keep `vertexColors` but transform it to boolean, can we keep a warning if we give something different than false/true?. I'm not sure if this is possible.

Reply:
My intention here was to keep back-compatibility on the schema as much as possible, so I didn't convert to a boolean: its a string that takes 2 values: "none" and "vertex".

We could change the property to a Boolean, but that would mean that applications written against the pre-1.4.1. API that used vertexColors would need updating to use the new API.

It depends whether you want to prioritize back compatibility, or prioritize having a cleaner API going forward?

Another option would be to have 2 properties on the schema:
- vertexColors, string, maintained for back-compatibility.  Deprecated, and retired at some point in the future
- vertexColorsEnabled, boolean.  Recommended to use this going forwards.

This last option probably gives best of both worlds - I'll refactor the PR based on this.

Reply:
@vincentfretin - my reply crossed with yours, so I'd not seen your comments when I wrote mine.  I've now updated the PR based on my proposal.  Happy to rework again if you can suggest a better way forwards.

Reply:
Your changes are ok to keep backward compatibility for now and to have a proper warning message to move to the other property, that's what matter I think.

Reply:
I looked into your suggestion, and we can convert vertexColors to a boolean, while still handling the string cases.

This does what we'd need.
```
material.vertexColors = !(!data.vertexColors || data.vertexColors === "none");
```
 +/- some code to generate an appropriate deprecation warning.
 
 So it comes down to whether we'd prefer the property name long-term to be vertexColors, or vertexColorsEnabled?

Reply:
I would rip the ban-aid and change to vertexColorsEnabled. Can perhaps through a deprecation message for those using the old property. `vertexColors` name being a boolean / on/off switch is misleading.

Reply:
OK, so are you happy with the PR in it's current state?

- vertexColors has to remain on the schema, if we are to deliver a bespoke deprecation warning.  If we remove it, users will just get a generic warning, which is more likely to lead to support tickets / queries.
- if it's left on the schema, I think it should also remain in docs (marked as deprecated).

Reply:
I would personally go ahead and change the name directly to vertexColorsEnabled and will make a note on the release notes.

Reply:
I think one consequence of that approach is more broken community components, which is already a problem that IMO is harming A-Frame adoption.

However I care more about having a fix for the bug that I do the back-compatibility, and there probably aren't very many community components that use material.vertexColors - so I'm OK to update this PR as you suggest.

Reply:
PR now updated - hopefully now mergeable

Reply:
> I think one consequence of that approach is more broken community components, which is already a problem that IMO is harming A-Frame adoption.

I hear you. A-Frame API has been pretty stable for a while. Backward compat issues are mainly a consequence of being on THREE train. Could lock A-Frame to an old THREE version or maintain own fork but thing those are worse and less sustainable solutions. I care about backwards compat but even more so about API usability, consistency and flexibility to evolve.

> However I care more about having a fix for the bug that I do the back-compatibility, and there probably aren't very many community components that use material.vertexColors - so I'm OK to update this PR as you suggest.

Thanks so much for the flexibility. Super appreciated. You rock.